examples/fluids/problems/stg_shur14.c

```diff
@@ -221,7 +221,7 @@ PetscErrorCode SetupStg(const MPI_Comm comm, const DM dm, ProblemData problem, U
   char                     stg_inflow_path[PETSC_MAX_PATH_LEN] = "./STGInflow.dat";
   char                     stg_rand_path[PETSC_MAX_PATH_LEN]   = "./STGRand.dat";
   PetscBool                mean_only = PETSC_FALSE, use_stgstrong = PETSC_FALSE, use_fluctuating_IC = PETSC_FALSE, given_stg_dx = PETSC_FALSE;
-  CeedScalar               u0 = 0.0, alpha = 1.01, stg_dx = 1.0e-3;
+  CeedScalar               u0 = 0.0, alpha = 1.01, stg_dx = 1.0e-3, stg_h_scale_factor = 1 / user->app_ctx->degree;
```

Comment:
So we query whether the `-stg_dx` flag as been used and replace the value with `domain_size[0] / face[0]` if it's not. So that `stg_dx` magic value is never used.

That being said, I do see the danger, but I'm not sure how to address it. I can set the value to something else (`0`, `-1`), but it's still a silent error; the code will still run, just incorrectly.

Reply:
Can we just not initialize `stg_dx` so we get a compiler warning if it isn't set? Or use `-1` as a placeholder and check somewhere that it has indeed been set? Just looking for something less error-prone.

Reply:
Sure, I've added a `PetscCheck` now to verify the size of `stg_dx`.

Reply:
Agreed and that is my bad.  I guess the two options I can think of (until we get 3D metric computation to directly compute) are:
1. error out if user does not set
2. use 1.5 times larger of the two in-plane metrics

Reply:
You're fine; pretty sure I'm the one who put that there.

The issue with 1 is that the default behavior has been to just go off of the box-mesh related quantities.
The issue with 2 is that it's an assumption (usually a good one, but an assumption none-the-less).

Note that this will error out if STG is not run on a PETSc-generated box mesh and `-stg_dx` isn't set. (the `-dm_plex_box_faces` default value is all zeros, so the calculated `stg_dx` will be infinitty and trigger the `PetscIsNormalReal` part of the `PetscCheck`.